esx_service/vmdk_ops.py

```diff
@@ -634,9 +645,10 @@ def disk_detach(vmdk_path, vm):
     dev_changes.append(disk_spec)
     spec.deviceChange = dev_changes
 
+    # Handle the most generic exception for a VM reconfig.
     try:
         wait_for_tasks(si, [vm.ReconfigVM_Task(spec=spec)])
-    except vim.fault.GenericVmConfigFault as ex:
+    except vim.fault.VimFault as ex:
```

Comment:
There is nothing that can be done about a VM changing state. The plugin can
at the most handle any errors that can cause it to crash. Other than that,
the plugin has no way to handle VM state changes.  If a VM is powering off
or reset or suspended then the VM reconfig will fail. The design for
handling configuration changes to the VM will have be a separate issue.
I'll raise that.

This issue is only going to prevent plugin-server exit because of unhandled
exception.

On Thu, Jun 2, 2016 at 12:30 AM, Prashant Dhamdhere <
notifications@github.com> wrote:

> In esx_service/vmdk_ops.py
> <https://github.com/vmware/docker-volume-vsphere/pull/433#discussion_r65422070>
> :
>
> >      try:
> >          wait_for_tasks(si, [vm.ReconfigVM_Task(spec=spec)])
> > -    except vim.fault.GenericVmConfigFault as ex:
> > +    except vim.fault.VimFault as ex:
>
> It seems consistent to change GenericVmConfigFault =>VimFault. However,
> this doesn't really fix #381
> <https://github.com/vmware/docker-volume-vsphere/issues/381>.
>
> Unrelated to your change but does logging.warning(f.message) logs
> exception code too?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/433/files/1ae935fbbc0c3205389f8fd83888b9afea83cba2#r65422070>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseJK3p3Iy1KN0JdaE5Rc6BDpk5411ks5qHdbAgaJpZM4IrROI>
> .
>